mlflow/types/llm.py

```diff
@@ -0,0 +1,226 @@
+        if not isinstance(values, list):
+            raise ValueError(f"`{key}` must be a list")
+
+        if len(values) > 0:
+            # if the items are all dicts, try to convert them to the desired class
+            if all(isinstance(v, dict) for v in values):
+                try:
+                    setattr(self, key, [cls(**v) for v in values])
+                except TypeError as e:
+                    raise ValueError(f"Error when coercing {values} to {cls.__name__}: {e}")
+            elif all(isinstance(v, cls) for v in values):
+                pass
+            else:
+                raise ValueError(
+                    f"Items in `{key}` must all have the same type: {cls.__name__} or dict"
+                )
+
+    def to_dict(self):
+        return asdict(self, dict_factory=lambda obj: {k: v for (k, v) in obj if v is not None})
+
+
+@dataclass
+class ChatMessage(_BaseDataclass):
+    """A message in a conversation."""
+
+    role: str
+    content: str
+    name: Optional[str] = None
+
+    def __post_init__(self):
+        self._validate_field("role", str, True)
+        self._validate_field("content", str, True)
+        self._validate_field("name", str, False)
+
+
+@dataclass
+class ChatParams(_BaseDataclass):
+    """Parameters for chat."""
+
+    temperature: float = 1.0
+    max_tokens: Optional[int] = None
+    stop: Optional[List[str]] = None
+    n: int = 1
+    stream: bool = False
+
+    def __post_init__(self):
+        self._validate_field("temperature", float, True)
+        self._validate_field("max_tokens", int, False)
+        self._validate_list("stop", str, False)
+        self._validate_field("n", int, True)
+        self._validate_field("stream", bool, True)
+
+
+@dataclass()
+class ChatRequest(ChatParams):
+    """A request to chat."""
+
+    messages: List[ChatMessage] = field(default_factory=list)
+
+    def __post_init__(self):
+        self._convert_dataclass_list("messages", ChatMessage)
+        super().__post_init__()
+
+
+@dataclass
+class ChatChoice(_BaseDataclass):
+    """A choice in a chat response."""
+
+    index: int
+    message: ChatMessage
+    finish_reason: str
+
+    def __post_init__(self):
+        self._validate_field("index", int, True)
+        self._validate_field("finish_reason", str, True)
+        if not isinstance(self.message, ChatMessage):
+            self.message = ChatMessage(**self.message)
+
+
+@dataclass
+class TokenUsageStats(_BaseDataclass):
+    """Stats about token usage."""
+
+    prompt_tokens: int
+    completion_tokens: int
+    total_tokens: int
+
+    def __post_init__(self):
+        self._validate_field("prompt_tokens", int, True)
+        self._validate_field("completion_tokens", int, True)
+        self._validate_field("total_tokens", int, True)
+
+
+@dataclass
+class ChatResponse(_BaseDataclass):
+    """A response from chat."""
+
+    model: str
+    choices: List[ChatChoice]
+    usage: TokenUsageStats
+    object: str = "chat.completion"
```

Comment:
```suggestion
    object: Literal["chat.completion"] = "chat.completion"
```

Can we use `Literal`?